docgenerator/SDKDocGenerator.UnitTests/ExampleMetadataParserTests.cs

```diff
@@ -0,0 +1,151 @@
+        }
+
+        [Fact]
+        public void GenerateExampleFragments_WithValidFile_ShouldCreateControlTowerFragment()
+        {
+            ExampleMetadataParser.GenerateExampleFragments(_testMetaJsonPath);
+            try
+            {
+                var fragmentPath = Path.Combine(ExampleMetadataParser.ExampleFragmentsFullPath, "ControlTower.fragment.html");
+                Assert.True(File.Exists(fragmentPath));
+
+                var content = File.ReadAllText(fragmentPath);
+                Assert.Contains("<a href=\"https://docs.aws.amazon.com/code-library/latest/ug/dotnet_4_controltower_code_examples.html\" target=\"_blank\">AWS SDK Code Examples Code Library</a>", content);
+                Assert.Contains("<h3 class=\"\">Api</h3>", content);
+                Assert.Contains("Use DisableBaseline", content);
+                Assert.Contains("Use DisableControl", content);
+                Assert.Contains("Use EnableBaseline", content);
+                Assert.Contains("Use EnableControl", content);
+                Assert.Contains("Use GetBaselineOperation", content);
+                Assert.Contains("Use GetControlOperation", content);
+                Assert.Contains("Use ListBaselines", content);
+                Assert.Contains("Use ListEnabledBaselines", content);
+                Assert.Contains("Use ListEnabledControls", content);
+                Assert.Contains("Use ListLandingZones", content);
+                Assert.Contains("Use ResetEnabledBaseline", content);                
+            }
+            finally
+            {
+                ExampleMetadataParser.CleanupExampleFragments();
+            }
+        }
+
+        [Fact]
+        public void GenerateExampleFragments_WithNonExistentFile_ShouldNotThrow()
+        {
+            var nonExistentPath = Path.Combine(Path.GetTempPath(), "nonexistent.json");
+            try
+            {
+                var exception = Record.Exception(() => ExampleMetadataParser.GenerateExampleFragments(nonExistentPath));
+                Assert.Null(exception);
+            }
+            finally
+            {
+                ExampleMetadataParser.CleanupExampleFragments();
+            }
+        }
+
+        [Fact]
+        public void CleanupExampleFragments_ShouldRemoveDirectory()
+        {
+            ExampleMetadataParser.GenerateExampleFragments(_testMetaJsonPath);
+            var fragmentsPath = ExampleMetadataParser.ExampleFragmentsFullPath;
+            try
+            {
+                Assert.True(Directory.Exists(fragmentsPath));
+            }
+            finally
+            {
+                ExampleMetadataParser.CleanupExampleFragments();
+            }
+            
+            Assert.False(Directory.Exists(fragmentsPath));
+        }
+
+        [Fact]
+        public void CleanupExampleFragments_WithNullPath_ShouldNotThrow()
+        {
+            var exception = Record.Exception(() => ExampleMetadataParser.CleanupExampleFragments());
```

Comment:
TIL about `Record.Exception`. I had been using `Assert.Throws<T>` but it requires the type to be specified...